woocommerce/woocommerce-android
# Issue/5923 product tags recycler crash

Fixes #5923 - I wasn't able to reproduce the crash, but we've run into this "Cannot call this method while RecyclerView is computing a layout or scrolling" issue before and the solution was to avoid using `notifyDatasetChanged()`.

In this specific crash, my guess is the problem was due to calling `notifyDatasetChanged()` when the product tags filter changes. This was unnecessary since the ViewModel takes care of filtering the tags, and the fragment observes the ViewModel's tags and updates the adapter when the tags change.

I also removed an unused `context` in the adapter's constructor.

To test:

* Open product detail for a product that has tags
* Tap "Tags"
* Apply a filter to the tags
* Verify that the tag list is correctly filtered
